indra/newview/llfloateremojipicker.cpp

```diff
@@ -37,11 +37,20 @@
 #include "llscrollingpanellist.h"
 #include "llscrolllistctrl.h"
 #include "llscrolllistitem.h"
+#include "llsdserialize.h"
 #include "lltextbox.h" 
 #include "llviewerchat.h" 
 
-size_t LLFloaterEmojiPicker::sSelectedGroupIndex;
-std::string LLFloaterEmojiPicker::sSearchPattern;
+static U32 sSelectedGroupIndex;
```

Comment:
From coding policy:

> Static class member variables start with an 's' followed by an uppercase character

This isn't a memeber variable, I think it can be defined as a filescoped global variable and should be marked as 'g', gSelectedGroupIndex.

Suggestion: it has no reason to be filscoped, just move declaration into class's header and mark as private.

Reply:
IMHO no needs to declare these variables in the class definition
No any other class or module should know about these variables

Global variables are variables accessible from any file or module
These variables aren't global, they are niot accessible from outside
Thus the prefix "g" would look strange or confusive in their names

Reply:
The prefix "s" is used for file-scoped variables as well
![image](https://github.com/secondlife/viewer/assets/124201357/5b510e9a-af7e-4401-ba12-9128f6e91ac0)

Reply:
>The prefix "s" is used for file-scoped variables as well
Check our coding policy document (sadly doesn't cover namespaces)

personally I have no issues with it remaining an 's', since it is a static, just pointing out that it is not according to coding policy.

>No any other class or module should know about these variables
If they will get marked as private, none would